mcp-aws-ec2-server/src/mcp_aws_ec2_server/main.py

```diff
@@ -230,14 +159,8 @@ async def search_instances_by_name(
     ] = "us-east-1",
 ):
     """Search for instances by name in the specified AWS region."""
-    ec2_client = boto3.client("ec2", region_name=aws_region)
-    response = ec2_client.describe_instances(
-        Filters=[{"Name": "tag:Name", "Values": [instance_name]}]
-    )
-    try:
-        return {"instances": response["Reservations"]}
-    except Exception as e:
-        return {"error": f"Failed to search instances by name: {str(e)}"}
+    service = get_aws_service(aws_region)
+    return service.search_instances_by_tag(tag_key="Name", tag_value=instance_name)
```

Comment:
This needs to call the `search_instances_by_name` function.

Reply:
It's intentionally this way, since name is just a tag in AWS as well. I'm on my phone right now so it's hard to look, but I believe I deleted that function. This tool is just meant to be a wrapper